composition-js/src/merging/merge.ts

```diff
@@ -248,6 +242,56 @@ function locationString(locations: DirectiveLocation[]): string {
   return (locations.length === 1 ? 'location ' : 'locations ') + '"' + locations.join(', ') + '"';
 }
 
+/**
+ * Given a list of directive sources, get the set of locations, removing duplicates.
+ * Executable locations are merged by intersection while non-executable locations are merged by union
+ */
+const getLocationsFromDirectiveDefs = (sources: (DirectiveDefinition | undefined)[]) => {
```

Comment:
**question** Is there a reason to use function expressions instead of function declarations? I see a few of the latter style already in this file, but not sure if it's already mixed. FWIW, I don't feel strongly.